src/Stratis.Bitcoin.Features.Consensus/ConsensusFeature.cs

```diff
@@ -135,8 +137,10 @@ public static IFullNodeBuilder UsePosConsensus(this IFullNodeBuilder fullNodeBui
                         services.AddSingleton<ConsensusController>();
                         services.AddSingleton<ConsensusStats>();
                         services.AddSingleton<IConsensusRuleEngine, PosConsensusRuleEngine>();
-                        services.AddSingleton<IGetUnspentTransaction, ConsensusQuery>();
-
+                        services.AddSingleton<IChainState, ChainState>();
+                        services.AddSingleton<ConsensusQuery>()
+                            .AddSingleton<INetworkDifficulty, ConsensusQuery>(provider => provider.GetService<ConsensusQuery>())
+                            .AddSingleton<IGetUnspentTransaction, ConsensusQuery>();
```

Comment:
I think you need to do the same here as well to get the CQ instance
```
  .AddSingleton<IGetUnspentTransaction, ConsensusQuery>(provider => provider.GetService<ConsensusQuery>())
```

Reply:
Done.

Reply:
@thefazzer I have made the change for you 👍